### ISSUE TITLE
allow external CSS/JavaScript selection from assets folder as well

### ISSUE THREAD

**Issue by fritzmg** [contao/core-bundle#1301]
Suppose you are installing CSS or JavaScript assets via composer. When using the contao/standard-edition or managed-edition these assets will be installed into the `/assets` folder.

However, in the page layout you can only select files within the `files` folder for external CSS or JavaScript files. I typically add these files then via a `js_assets.html5` template where I insert something like
```php
<?php $GLOBALS['TL_CSS'][] = 'assets/font-awesome/css/font-awesome.min.css'; ?>
```
for example.

But wouldn't it make sense to be able to select external CSS and JavaScript files from the assets directory as well?

**Comment by Toflar** [comment]
We should just allow everything that's in the `web` folder. It would include `files` and everything that's linked there publicly ;-)

**Comment by aschempp** [comment]
I disagree. The picker picks a file from the manager (by linking a UUID). For everything else, just add custom header or script tags.

**Comment by frontendschlampe** [comment]
It would be great if we even could choose a js-file with a file picker like a css-file ;-)

Theme+ had this functionality and it was great. There were also some other useful things included in Theme+:

- include file in head or body
- merge files to one
- global css/js picker in themes and assignment to layouts
- in file picker you can switch between files, assets or other folder
and some more

Maybe there will be some ideas to include in core

**Comment by leofeyer** [comment]
> It would be great if we even could choose a js-file with a file picker like a css-file ;-)

This is already implemented in Contao 4.5.

> The picker picks a file from the manager (by linking a UUID). For everything else, just add custom header or script tags.

👍
